scribejava-apis/src/main/java/com/github/scribejava/apis/StackExchangeApi.java

```diff
@@ -0,0 +1,59 @@
+package com.github.scribejava.apis;
+
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.model.OAuthConfig;
+import com.github.scribejava.core.model.OAuthConstants;
+import com.github.scribejava.core.model.Verb;
+import com.github.scribejava.core.utils.OAuthEncoder;
+import com.github.scribejava.core.utils.Preconditions;
+
+/***
+ * Stack Exchange authentication via OAuth 2.0  (stackoverflow.com,
+ * askubuntu.com, etc.).
+ *
+ * @author Michal Foksa
+ *
+ */
+public class StackExchangeApi extends DefaultApi20 {
+
+    private static final String AUTHORIZE_URL
+    = "https://stackexchange.com/oauth?client_id=%s&redirect_uri=%s";
+
+    private StackExchangeApi() {
```

Comment:
make it protected. It was my fault to make all API constructors private. I fixed that. In master branch all API constructors are protected, so anyone can extend API in their code if they want. But couldn't instantiate directly, forcing them to use singleton